storage/accumulator/src/lib.rs

```diff
@@ -318,9 +317,6 @@ where
     ///     and the full route from root of that subtree to the accumulator root turns frozen
     ///         height - (log2(num_new_leaves) + 1) < height - 1 = root_level
     fn max_to_freeze(num_new_leaves: usize, root_level: u32) -> usize {
-        precondition!(root_level as usize <= MAX_ACCUMULATOR_PROOF_DEPTH);
```

Comment:
...but add `debug_asserts!` !!!!!

Reply:
No?

Reply:
They didn't justify a runtime assert back then (probably more here just for mirai people having fun). I tend to agree we can just remove them. Probably have a second look case by case, but definitely not replacing every one of them an assertion.

Reply:
Not a runtime assertion, but why not a debug assertion?

Reply:
Too verbose? 

Also it's like overflow-check we were discussing, if we check it in debug build, we should check it in release.

Reply:
But, AFAICT, this is not like the overflow check at all: this is just a good sanity check for mis-uses of this function. When we are debugging an issue in debug mode, this assert might save you a lot of trouble identifying the culprit.

Reply:
debug_assertion is for too expensive checking in release mode. If it is important to check and not expensive, we should use `assert!` instead.

Reply:
That's why we can go case eny case: to identify if it's worth nothing, a comment or an assertion. 

But to move faster I'm okay with making them all assertions as well.

Reply:
I agree with @lightmark that debug asserts are useful for avoiding expensive checks in release mode.

I also see @davidiw's point about the inconsistencies between dev and prod...

Not sure what the right path is here, so deferring to you.

Reply:
What I'd like to see is some solid thinking around what should and shouldn't be feasible and then some fuzzing that ensures that those expectations are met.

Reply:
Isn't the first half of that called formal verification? 😛